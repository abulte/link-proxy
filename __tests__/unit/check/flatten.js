const {flatten} = require('../../../jobs/check/flatten')

describe('check.flatten', () => {
  it('should return the file for a simple document', () => {
    const res = flatten({
      url: 'http://foo',
      type: 'file',
      fileTypes: [{ext: 'txt'}]
    })

    expect(res).toMatchSnapshot()
  })

  it('should flatten a shapefile in an index-of', () => {
    const res = flatten({
      url: 'http://foo',
      type: 'index-of',
      children: [
        {
          url: 'http://foo/1.shp',
          type: 'file',
          fileName: 'foo.shp',
          fileTypes: [{ext: 'shp'}]
        },
        {
          url: 'http://foo/1.shx',
          type: 'file',
          fileName: 'foo.shx',
          fileTypes: [{ext: 'shx'}]
        }
      ]
    })

    expect(res).toMatchSnapshot()
  })

  it('should return a sole shapefile in an index-of', () => {
    const res = flatten({
      url: 'http://foo',
      type: 'index-of',
      children: [
        {
          url: 'http://foo/1.shp',
          type: 'file',
          fileName: 'foo.shp',
          fileTypes: [{ext: 'shp'}]
        }
      ]
    })

    expect(res).toMatchSnapshot()
  })

  it('should not consider related files as main when they also have the main type', () => {
    const res = flatten({
      url: 'http://foo',
      type: 'index-of',
      children: [
        {
          url: 'http://foo/file.tiff',
          type: 'file',
          fileName: 'file.tiff',
          fileTypes: [{ext: 'tiff'}]
        },
        {
          url: 'http://foo/file.ovr',
          type: 'file',
          fileName: 'file.ovr',
          fileTypes: [{ext: 'ovr'}, {ext: 'tiff'}]
        }
      ]
    })

    expect(res).toMatchSnapshot()
  })

  it('should return multiple bundles with similar extensions', () => {
    const res = flatten({
      url: 'http://foo',
      type: 'index-of',
      children: [
        {
          url: 'http://foo/file1.shp',
          type: 'file',
          fileName: 'file1.shp',
          fileTypes: [{ext: 'shp'}]
        },
        {
          url: 'http://foo/file1.dbf',
          type: 'file',
          fileName: 'file1.dbf',
          fileTypes: [{ext: 'dbf'}]
        },
        {
          url: 'http://foo/file2.dbf',
          type: 'file',
          fileName: 'file2.dbf',
          fileTypes: [{ext: 'dbf'}]
        }
      ]
    })

    expect(res).toMatchSnapshot()
  })
})
