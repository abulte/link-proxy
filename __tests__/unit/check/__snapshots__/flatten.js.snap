// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`check.flatten should flatten a shapefile in an index-of 1`] = `
Object {
  "links": Object {
    "http://foo": Object {
      "bundles": Array [
        Object {
          "changed": 2,
          "files": Array [
            Object {
              "fileName": "foo.shp",
              "fileTypes": Array [
                Object {
                  "ext": "shp",
                },
              ],
              "type": "file",
              "url": "http://foo/1.shp",
            },
            Object {
              "fileName": "foo.shx",
              "fileTypes": Array [
                Object {
                  "ext": "shx",
                },
              ],
              "type": "file",
              "url": "http://foo/1.shx",
            },
          ],
          "type": "shp",
        },
      ],
      "errors": Array [],
      "urls": Array [
        "http://foo/1.shp",
        "http://foo/1.shx",
      ],
      "warnings": Array [],
    },
    "http://foo/1.shp": Object {
      "bundles": Array [],
      "errors": Array [],
      "urls": Array [],
      "warnings": Array [],
    },
    "http://foo/1.shx": Object {
      "bundles": Array [],
      "errors": Array [],
      "urls": Array [],
      "warnings": Array [],
    },
  },
  "temporaries": Array [],
}
`;

exports[`check.flatten should not consider related files as main when they also have the main type 1`] = `
Object {
  "links": Object {
    "http://foo": Object {
      "bundles": Array [
        Object {
          "changed": 2,
          "files": Array [
            Object {
              "fileName": "file.tiff",
              "fileTypes": Array [
                Object {
                  "ext": "tiff",
                },
              ],
              "type": "file",
              "url": "http://foo/file.tiff",
            },
            Object {
              "fileName": "file.ovr",
              "fileTypes": Array [
                Object {
                  "ext": "ovr",
                },
                Object {
                  "ext": "tiff",
                },
              ],
              "type": "file",
              "url": "http://foo/file.ovr",
            },
          ],
          "type": "tiff",
        },
      ],
      "errors": Array [],
      "urls": Array [
        "http://foo/file.tiff",
        "http://foo/file.ovr",
      ],
      "warnings": Array [],
    },
    "http://foo/file.ovr": Object {
      "bundles": Array [],
      "errors": Array [],
      "urls": Array [],
      "warnings": Array [],
    },
    "http://foo/file.tiff": Object {
      "bundles": Array [],
      "errors": Array [],
      "urls": Array [],
      "warnings": Array [],
    },
  },
  "temporaries": Array [],
}
`;

exports[`check.flatten should return a sole shapefile in an index-of 1`] = `
Object {
  "links": Object {
    "http://foo": Object {
      "bundles": Array [],
      "errors": Array [],
      "urls": Array [
        "http://foo/1.shp",
      ],
      "warnings": Array [],
    },
    "http://foo/1.shp": Object {
      "bundles": Array [
        Object {
          "changed": 1,
          "files": Array [
            Object {
              "fileName": "foo.shp",
              "fileTypes": Array [
                Object {
                  "ext": "shp",
                },
              ],
              "type": "file",
              "url": "http://foo/1.shp",
            },
          ],
          "type": "shp",
        },
      ],
      "errors": Array [],
      "urls": Array [],
      "warnings": Array [],
    },
  },
  "temporaries": Array [],
}
`;

exports[`check.flatten should return multiple bundles with similar extensions 1`] = `
Object {
  "links": Object {
    "http://foo": Object {
      "bundles": Array [
        Object {
          "changed": 2,
          "files": Array [
            Object {
              "fileName": "file1.shp",
              "fileTypes": Array [
                Object {
                  "ext": "shp",
                },
              ],
              "type": "file",
              "url": "http://foo/file1.shp",
            },
            Object {
              "fileName": "file1.dbf",
              "fileTypes": Array [
                Object {
                  "ext": "dbf",
                },
              ],
              "type": "file",
              "url": "http://foo/file1.dbf",
            },
          ],
          "type": "shp",
        },
      ],
      "errors": Array [],
      "urls": Array [
        "http://foo/file1.shp",
        "http://foo/file1.dbf",
        "http://foo/file2.dbf",
      ],
      "warnings": Array [],
    },
    "http://foo/file1.dbf": Object {
      "bundles": Array [],
      "errors": Array [],
      "urls": Array [],
      "warnings": Array [],
    },
    "http://foo/file1.shp": Object {
      "bundles": Array [],
      "errors": Array [],
      "urls": Array [],
      "warnings": Array [],
    },
    "http://foo/file2.dbf": Object {
      "bundles": Array [
        Object {
          "changed": 1,
          "files": Array [
            Object {
              "fileName": "file2.dbf",
              "fileTypes": Array [
                Object {
                  "ext": "dbf",
                },
              ],
              "type": "file",
              "url": "http://foo/file2.dbf",
            },
          ],
          "type": "dbf",
        },
      ],
      "errors": Array [],
      "urls": Array [],
      "warnings": Array [],
    },
  },
  "temporaries": Array [],
}
`;

exports[`check.flatten should return the file for a simple document 1`] = `
Object {
  "links": Object {
    "http://foo": Object {
      "bundles": Array [
        Object {
          "changed": 1,
          "files": Array [
            Object {
              "fileTypes": Array [
                Object {
                  "ext": "txt",
                },
              ],
              "type": "file",
              "url": "http://foo",
            },
          ],
          "type": "document",
        },
      ],
      "errors": Array [],
      "urls": Array [],
      "warnings": Array [],
    },
  },
  "temporaries": Array [],
}
`;
